File: backend/src/main/java/com/dresscode/service/UserService.java
package com.dresscode.service;

import java.util.List;
import java.util.Optional;

import com.dresscode.dto.user.AdminUserCreationRequestDto;
import com.dresscode.dto.user.UserUpdateRequestDto;
import com.dresscode.model.User;

public interface UserService {
    public List<User> getAllUsers();

    public List<User> getUsersByActive(boolean flag);

    public Optional<User> getUserById(Long id);

    public User createUser(User User);

    public User adminUserCreation(AdminUserCreationRequestDto dto);

    public User updateUser(UserUpdateRequestDto User, Long id);

    public Boolean toggleUserStatus(Long id);

    public void deleteUser(Long id);

}

File: frontend/src/pages/private/admin/service/userService.js
import ApiConfig from '../../../../api/ApiConfig'

const userService = {
  fetchUsers: async () => {
    return await ApiConfig.getAllUsers()
  },

  createUser: async (userData) => {
    return await ApiConfig.adminUserCreation(userData)
  },

  updateUser: async (id, userData) => {
    return await ApiConfig.updateUser(id, userData)
  },

  toggleUserStatus: async (id) => {
    return await ApiConfig.toggleUserStatus(id)
  },

  deleteUser: async (id) => {
    return await ApiConfig.deleteUser(id)
  },
}

export default userService
